### ISSUE TITLE
Support for XState v5 `guard` transition property

### ISSUE THREAD

**Issue by ts-candide** [rlaffers/eslint-plugin-xstate#27]
**Describe the bug**

XState v5 has renamed the `cond` property to `guard` for guarded transitions. This plugin reports `guard` as an invalid transition property when placed inside an `invoke.onDone` array.

**Expected behavior**

No error when using `guard` as a transition property inside an `invoke.onDone` array.

**Actual behavior**

An ESLint error is shown when using `guard` as a transition property inside an `invoke.onDone` array.

**Versions (please complete the following information):**
 - Node version: 20.9.0
 - ESLint version: 8.55.0
 - eslint-plugin-xstate version: 1.1.3
 - xstate version: 5.1.0

**Additional context**

Here is some sample machine config:

``` js
{
  states: {
    selectingReader: {
      invoke: {
        src: "startDiscoveringReaders",
        onDone: [
          {
            guard: "noReadersDiscovered",
            target: "noReadersFound",
          },
        ],
        onError: "errored",
      },
    }
  }
}
```

The `guard` property is being marked as invalid here.

<img width="587" alt="Screenshot 2023-12-07 at 14 43 16" src="https://github.com/rlaffers/eslint-plugin-xstate/assets/145444954/d0956fca-6584-4ebc-a1a4-e28170db16b0">

**Comment by rlaffers** [comment]
Thanks for the report, I will look at it soon (ish).

**Comment by rlaffers** [comment]
@ts-candide 
The version you are using is not compatible with xstate v5. Please use eslint-plugin-xstate@next.

**Comment by rlaffers** [comment]
Version 3.2.1 of this plugin now supports xstate v5.
